### PR TITLE
Add surface global context

### PR DESCRIPTION
**Pull request by TheNeikos**

### CODE REVIEW

**Comment by neithernut** [comment]
Targets #422.

**Comment by neithernut** [comment]
Please rebase.

**Comment by neithernut** [comment]
I'm not sure whether the syle fixes shouldn't be a commit on it's own...
But... ACK

File: src/context.c [hunk 1]
```diff
@@ -298,3 +317,25 @@ func_get_surface_under_cursor(
     return 0;
 }
 
+static int
+func_get_ms_focus(
+    union ws_value_union* stack
+) {
+    struct ws_surface* surface = ws_cursor_get()->active_surface;
```
Comment (matthiasbeyer) [inline src/context.c:324]: It is guaranteed that the `ws_cursor_get()` always returns the expected thing?

File: src/context.c [hunk 2]
```diff
@@ -298,3 +317,25 @@ func_get_surface_under_cursor(
...
+    ws_value_object_id_set(&stack[0].object_id, (struct ws_object*) surface);
+
+    return 0;
+}
+
+static int
+func_get_kb_focus(
+    union ws_value_union* stack
+) {
+    struct ws_surface* surface = ws_keyboard_get()->active_surface;
```
Comment (matthiasbeyer) [inline src/context.c:335]: Same here.

File: src/context.c [hunk 3]
```diff
@@ -298,3 +317,25 @@ func_get_surface_under_cursor(
     return 0;
 }
 
+static int
+func_get_ms_focus(
+    union ws_value_union* stack
+) {
+    struct ws_surface* surface = ws_cursor_get()->active_surface;
+
+    ws_value_object_id_set(&stack[0].object_id, (struct ws_object*) surface);
```
Comment (matthiasbeyer) [inline src/context.c:326]: No need to reinit the stack position before?

File: src/context.c [hunk 4]
```diff
@@ -298,3 +317,25 @@ func_get_surface_under_cursor(
...
+    return 0;
+}
+
+static int
+func_get_kb_focus(
+    union ws_value_union* stack
+) {
+    struct ws_surface* surface = ws_keyboard_get()->active_surface;
+
+    ws_value_object_id_set(&stack[0].object_id, (struct ws_object*) surface);
```
Comment (matthiasbeyer) [inline src/context.c:337]: No need to reinit the stack position before?

File: src/context.c (hunk 1 above)
Comment (TheNeikos) [inline src/context.c:324]: Yes, it is just a getter for an internal struct

File: src/context.c (hunk 3 above)
Comment (TheNeikos) [inline src/context.c:326]: No as it is already an object id.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**